### ISSUE TITLE
`wg-quick: line 85: exec: sudo: not found`

### ISSUE THREAD

**Issue by 9SMTM6** [linuxserver/docker-wireguard#91]
[linuxserverurl]: https://linuxserver.io
[![linuxserver.io](https://raw.githubusercontent.com/linuxserver/docker-templates/master/linuxserver.io/img/linuxserver_medium.png)][linuxserverurl]

Otherwise ignored error on aarch64 (Arch-ARM64): `wg-quick: line 85: exec: sudo: not found`
------------------------------

## Expected Behavior

Be able to connect to the server with a client and use the VPN to access both the general internet as well as local servers

## Current Behavior

I'm perfectly able to scan the QR Codes with my clients, the config in the client seems fine, and when I connect I get no error, but I cant use the internet anymore and I recieve 0 bytes even when trying to use the internet. 

## Steps to Reproduce

1. Install [Manjaro KDE on Raspberry Pi4](https://manjaro.org/download/#raspberry-pi-4-kde-plasma)
2. Install docker-compose etc.
3. Install `wireguard-dkms` kernel-module and for good measure `wireguard-tools` from the manjaro repos (pretty much arch repos) (take note, I did not manually install any `kernel-header` or similar)
4. Start this docker container using (take note of the different mapping for the kernel modules `/usr/lib/modules:/lib/modules`):
```yaml
version: "3.7"
services:
  wireguard:
    cap_add:
    - NET_ADMIN
    - SYS_MODULE
    container_name: wireguard
    environment:
      PEERS: phone,ipad,laptop,guest
      PGID: '1000'
      PUID: '1000'
      SERVERPORT: '51820'
      SERVERURL: *********
      TZ: europe/berlin
    image: linuxserver/wireguard # also tried with :v1.0.20200827-ls10
    ports:
    - mode: host
      protocol: udp
      published: 51820
      target: 51820
    restart: unless-stopped
    sysctls:
      net.ipv4.conf.all.src_valid_mark: '1'
    volumes:
    - /**/wireguard:/config:rw
    - /usr/lib/modules:/lib/modules:rw
    - /usr/src:/usr/src:rw
```
5. Try without success to use the VPN
6. Look at log (this is after repeated attempts so no new client configs were generated):
```
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] 01-envfile: executing... 
[cont-init.d] 01-envfile: exited 0.
[cont-init.d] 10-adduser: executing... 

-------------------------------------
          _         ()
         | |  ___   _    __
         | | / __| | |  /  \ 
         | | \__ \ | | | () |
         |_| |___/ |_|  \__/


Brought to you by linuxserver.io
-------------------------------------

To support the app dev(s) visit:
WireGuard: https://www.wireguard.com/donations/

To support LSIO projects visit:
https://www.linuxserver.io/donate/
-------------------------------------
GID/UID
-------------------------------------

User uid:    1000
User gid:    1000
-------------------------------------

[cont-init.d] 10-adduser: exited 0.
[cont-init.d] 30-config: executing... 
Uname info: Linux 83ca222593cc 5.4.83-1-MANJARO-ARM #1 SMP PREEMPT Mon Dec 14 08:37:33 CST 2020 aarch64 aarch64 aarch64 GNU/Linux
**** It seems the wireguard module is already active. Skipping kernel header install and module compilation. ****
**** Server mode is selected ****
**** External server address is set to 84.130.197.238 ****
**** External server port is set to 51820. Make sure that port is properly forwarded to port 51820 inside this container ****
**** Internal subnet is set to 10.13.13.0 ****
**** AllowedIPs for peers 0.0.0.0/0, ::/0 ****
**** PEERDNS var is either not set or is set to "auto", setting peer DNS to 10.13.13.1 to use wireguard docker host's DNS. ****
**** Server mode is selected ****
**** No changes to parameters. Existing configs are used. ****
[cont-init.d] 30-config: exited 0.
[cont-init.d] 99-custom-scripts: executing... 
[custom-init] no custom files found exiting...
[cont-init.d] 99-custom-scripts: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
/usr/bin/wg-quick: line 85: exec: sudo: not found
.:53
CoreDNS-1.8.1
linux/arm64, go1.15.7, 95622f4
```
7. Spot the otherwise easy to overlook error `/usr/bin/wg-quick: line 85: exec: sudo: not found` and wonder whether this is the issue

## Environment
**OS:** Manjaro KDE Raspberry Pi 4
**CPU architecture:** arm64
**How docker service was installed:** see above (hub.docker.com)

## Command used to create docker container (run/create/compose/screenshot)

see above

## Docker logs

see above

**Comment by 9SMTM6** [comment]
As noted above I did also check out an older version (`v1.0.20200827-ls10`) and got the same error.

I also just tried running `wg-quick up wg0` on metal after copying the config generated by this docker container into `/etc/wireguard/wg0.conf`. Seemed to work fine, I didnt try to connect but it didnt spit out the same error as above but:
```
#] ip link add wg0 type wireguard
[#] wg setconf wg0 /dev/fd/63
[#] ip -4 address add 10.13.13.1 dev wg0
[#] ip link set mtu 1420 up dev wg0
[#] ip -4 route add 10.13.13.9/32 dev wg0
[#] ip -4 route add 10.13.13.8/32 dev wg0
[#] ip -4 route add 10.13.13.7/32 dev wg0
[#] ip -4 route add 10.13.13.6/32 dev wg0
[#] iptables -A FORWARD -i wg0 -j ACCEPT; iptables -A FORWARD -o wg0 -j ACCEPT; iptables -t nat -A POSTROUTING -o eth0 -j MASQUERADE
```

**Comment by aptalca** [comment]
That's a straight up wireguard error that I have not seen before. Try installing sudo and see what happens:
`docker exec -it wireguard /bin/sh -c 'apt update && apt install sudo'`

**Comment by aptalca** [comment]
This is line 85 if the wg-quick: https://git.zx2c4.com/wireguard-tools/tree/src/wg-quick/linux.bash#n85

It is only executed if it is not running as uid 0. Are you doing rootless docker or something like that? The docker daemon (and the container) needs to run as root (0). It seems there is an issue with your docker install.

**Comment by 9SMTM6** [comment]
> That's a straight up wireguard error that I have not seen before. Try installing sudo and see what happens:
> `docker exec -it wireguard /bin/sh -c 'apt update && apt install sudo'`

This seems to fix that particular issue. I still cant get a proper connection but I dont get that error anymore.

> This is line 85 of the wg-quick script: https://git.zx2c4.com/wireguard-tools/tree/src/wg-quick/linux.bash#n85
> 
> It is only executed if it is not running as uid 0. Are you doing rootless docker or something like that? The docker daemon (and the container) needs to run as root (0). It seems there is an issue with your docker install.

I've not added myself to the docker-group but run docker(-compose) with sudo. I just checked with `sudo su`, the UID of my root user is 0.

On a few other containers I proxy the docker-socket, but as this container doesn't connect to the docker socket the only thing left over is a env-variable `DOCKER_HOST: tcp://socket-proxy:2375` that I add in some env-file I include in all my docker-config services. I also typically run `privileged: false` and `security_opt: - no-new-privileges:true`, but I've removed that after I got these kind of errors, it behaved the same.

I will however note that as far as I can see the `wg-quick up wg0` command is run by the container as `UID=1000` - which is my `UID` on metal - , as thats the `UID` I've set as env-variable on the container (as I do with all my other containers, including other linuxserver ones).

**Comment by 9SMTM6** [comment]
I've just checked what happens if I'm in the `docker` group. Same error.

**Comment by 9SMTM6** [comment]
Just to clarify, I set the PUID and PGID just as seen in the docker-compose file above and as recommended in your Readme. 

And while the error goes away the issues I face afterwards seem of a different kind (seems something to do with the DNS and Internet Subnet and/or Allowed IP), and also that error of course returns every time the container gets built again from a fresh image, I. E. Every time I change a parameter in the docker-compose that causes recreation of the service.

**Comment by 9SMTM6** [comment]
É

**Comment by aptalca** [comment]
> I will however note that as far as I can see the wg-quick up wg0 command is run by the container as UID=1000 - which is my UID on metal - , as thats the UID I've set as env-variable on the container (as I do with all my other containers, including other linuxserver ones).

That is not the case, wg-quick is meant to run as root, independent of the PUID/PGID set.

> I've not added myself to the docker-group but run docker(-compose) with sudo.

> I've just checked what happens if I'm in the docker group. Same error, even without the security options.

These have nothing to do with which user docker runs the container as. They only affect which user sends the commands via docker cli to the docker daemon.

> On a few other containers I proxy the docker-socket, but as this container doesn't connect to the docker socket the only thing left over is a env-variable DOCKER_HOST: tcp://socket-proxy:2375 that I add in some env-file I include in all my docker-config services. I also typically run privileged: false and security_opt: - no-new-privileges:true, but I've removed that after I got these kind of errors, it behaved the same.

I have a feeling you made more changes than you let on or maybe remember, that are causing this issue with the container not running as the root (0) user.

**Comment by 9SMTM6** [comment]
I was aware that all these things most likely don't have anything to do with the problem. Its just I'm not certain of that, and in an effort to avoid what you are implying here:

> I have a feeling you made more changes than you let on or maybe remember

I just threw all that information at you.

But, while writing this response I've indeed found the issue. I included a basics.env `env_file`. In that file I have been setting - among other things - the `$PGID` and `$PUID`. But in a mistake I started by setting the `$GID` and `$UID`, only to correct it later when they didnt work. In that step I added the `$PGID` and `$PUID` first to see if they work. They did and I removed the `$GID` and `$UID`, or so I thought. I guess I didnt save the file?

Dunno how I missed that, sorry to have taken up your time.
